### DISCUSSION TITLE
Working with coordinates

### DISCUSSION THREAD

**Discussion by nk-alex** [empira/PDFsharp#139] [Q&A] [answered]
This was originally asked as an issue (#138)

Having a class like this:

```
public class Element
{
	public int Left {get; set;}
	public int Top {get; set;}
	public int Height {get; set;}
	public int Width {get; set;}
	public String Value {get; set;}
}
```

I have a list of Element I would like to show as tables in a Document. Also, I would like to place them in the given coordinates (Left, Top, Height, Width). Units are millimeters.

This is how I'm trying to do it:

```
var document = new Document
{
    Info =
    {
        Title = "Title",
        Subject = "Subject",
        Author = "nk-alex"
    }
};

Section section = document.LastSection;
TextFrame frame = section.AddTextFrame();

List<Element> Elements = ...;
foreach(Element element in Elements)
{
	frame.MarginLeft = Unit.FromMillimeter(element.Left);
	frame.MarginTop = Unit.FromMillimeter(element.Top);

	 var table = new Table
	 {
	     Borders =
	     {
	         Width = 0.75
	     }
	 };

	var column = table.AddColumn(Unit.FromMillimeter(element.Width));
	column.Format.Alignment = ParagraphAlignment.Center;
	
	var row = table.AddRow();
	row.Format.Alignment = ParagraphAlignment.Center;
	row.Height = Unit.FromMillimeter(element.Height);

	var cell = row.Cells[0];
	cell.AddParagraph(element.Value);
	frame.Add(table);
}
```

But the result is not quite what I'm looking for:

![image](https://github.com/user-attachments/assets/169b41cc-11ca-4dec-bd07-d5dc7fafa2ef)

The first 6 cells showing on document, should be showing on the same row, but instead, they are showing one on top of the other.

What am I doing wrong? I want every element to be drawn as a separate table, so I'm not considering drawing them as columns of the same table.

@ThomasHoevel maybe you can give me some advice

**Comment by TH-Soft** [comment, marked as answer]
You have to set `.WrapFormat.Style = WrapStyle.Through;` for the TextFrame. And maybe play a little bit with the other WrapFormat properties.

If you download the PDFsharp repo and the assets, check out file `TextFrame-Layout.pdf` in folder `assets\archives\grammar-by-example\GBE\ReferencePDFs\GDI 1.40` for more information.
